citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/ConceptpowerImporter.java

```diff
@@ -0,0 +1,158 @@
+            throws URISyntaxException, AuthorityServiceConnectionException {
+        return null;
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize)
+            throws AuthorityServiceConnectionException {
+
+        RequestEntity<Void> request;
+        try {
+
+            String url = conceptpowerURL + conceptpowerSearchKeyword + URLEncoder
+                    .encode(this.getConceptpowerSearchString(firstName, lastName), StandardCharsets.UTF_8.toString())
+                    + "&page=" + page;
+            URI uri = UriComponentsBuilder.fromUriString(url.toString()).build(true).toUri();
+
+            request = RequestEntity.get(uri).accept(MediaType.APPLICATION_JSON).build();
+        } catch (UnsupportedEncodingException e) {
+            throw new AuthorityServiceConnectionException("Exception occured while creating URI for authority search",
+                    e);
+        }
+        ResponseEntity<ConceptpowerResponse> response = null;
+        try {
+            response = restTemplate.exchange(request, ConceptpowerResponse.class);
+        } catch (RestClientException ex) {
+            throw new AuthorityServiceConnectionException(ex);
+        }
+
+        List<IAuthorityEntry> authorityEntries = new ArrayList<IAuthorityEntry>();
+        AuthoritySearchResult searchResult = new AuthoritySearchResult();
+        if (response.getStatusCode() == HttpStatus.OK) {
+            ConceptpowerResponse conceptEntries = response.getBody();
+            if (conceptEntries.getConceptEntries() != null) {
+                for (ConceptpowerEntry conceptEntry : conceptEntries.getConceptEntries()) {
+                    IAuthorityEntry authority = new AuthorityEntry();
+                    authority.setName(conceptEntry.getLemma());
+                    authority.setUri(conceptEntry.getConcept_uri());
+                    authority.setDescription(conceptEntry.getDescription());
+                    authorityEntries.add(authority);
+                }
+                searchResult.setFoundAuthorities(authorityEntries);
+                searchResult.setTotalPages((int) Math
+                        .ceil(conceptEntries.getPagination().getTotalNumberOfRecords() / new Float(pageSize)));
+            }
+        } else {
+            throw new AuthorityServiceConnectionException(response.getStatusCode().toString());
+        }
+        return searchResult;
+    }
+
+    private String getConceptpowerSearchString(String firstName, String lastName) {
+
+        String conceptpowerSearchString = "";
+
+        if (firstName != null && lastName != null) {
+
+            conceptpowerSearchString = "%" + firstName + "%" + lastName;
+
+        } else {
+
+            conceptpowerSearchString = firstName == null ? conceptpowerSearchKeyword + "%" + lastName : "%" + firstName;
```

Comment:
in `conceptpowerSearchKeyword + "%" + lastName`, is `conceptpowerSearchKeyword` needed? wouldn't that always be empty string?

Reply:
also, what are the `%` for?